cedar-policy-mcp-schema-generator/src/description/deserializer.rs

```diff
@@ -0,0 +1,480 @@
+                    "Expected `description` attribute to be a string.",
+                    ContentType::Property,
+                )
+            })
+        })
+        .transpose()?;
+    Ok(Property::new(
+        name,
+        required,
+        property_type_from_json_value(json_value)?,
+        description,
+    ))
+}
+
+fn property_type_from_json_value(
+    json_value: &LocatedValue,
+) -> Result<PropertyType, DeserializationError> {
+    if let Some(ptype_obj) = json_value.get_object() {
+        if let Some(type_json) = ptype_obj.get("type") {
+            match type_json.get_str() {
+                Some("boolean") => Ok(PropertyType::Bool),
+                Some("integer") => Ok(PropertyType::Integer),
+                Some("float") => Ok(PropertyType::Float),
+                Some("number") => Ok(PropertyType::Number),
+                Some("string") => {
+                    if let Some(enum_json) = ptype_obj.get("enum") {
+                        if let Some(enum_variants) = enum_json.get_array() {
+                            let variants = enum_variants.iter().map(|variant| {
+                                variant.get_smolstr().ok_or_else(|| DeserializationError::unexpected_type(
+                                    variant,
+                                    "Expected element of `enum` attribute to be a string.",
+                                    ContentType::PropertyType
+                                ))
+                            }).collect::<Result<Vec<_>,_>>()?;
+                            if variants.is_empty() {
+                                Err(DeserializationError::unexpected_value(
+                                    enum_json,
+                                    "Expected non-empty list of variants for `enum` attribute.",
+                                    ContentType::PropertyType
+                                ))
+                            } else {
+                                Ok(PropertyType::Enum { variants })
+                            }
+                        } else {
+                            Err(DeserializationError::unexpected_type(
+                                enum_json,
+                                "Expected `enum` attributed to be a JSON array of strings.",
+                                ContentType::PropertyType
+                            ))
+                        }
+                    } else if let Some(format_json) = ptype_obj.get("format") {
+                        match format_json.get_str() {
+                            Some("date") => Ok(PropertyType::Datetime),
+                            Some("date-time") => Ok(PropertyType::Datetime),
+                            Some("duration") => Ok(PropertyType::Duration),
+                            Some("ipv4") => Ok(PropertyType::IpAddr),
+                            Some("ipv6") => Ok(PropertyType::IpAddr),
+                            Some("decimal") => Ok(PropertyType::Decimal),
```

Comment:
What would happen if we get an arbitrary floating number? Decimal cannot express all decimal values.